src/docbuild/cli/cmd_cli.py

```diff
@@ -136,12 +141,25 @@ def cli(
         None,
         DEFAULT_APP_CONFIG,
     )
-    context.appconfig = replace_placeholders(context.appconfig)
 
-    # Phase 2: Advanced logging setup with user configuration.
-    logging_config = context.appconfig.get('logging', {})
+    # 2. Validate the raw config dictionary using Pydantic
+    try:
+        # Pydantic validation also handles placeholder replacement via @model_validator
+        context.appconfig = AppConfig.from_dict(raw_appconfig)
+    except (ValueError, ValidationError) as e:
+        log.error("Application configuration failed validation:")
+        log.error(f"Error in config file(s): {context.appconfigfiles}")
```

Comment:
Remember lazy logging? 😉 


```suggestion
        log.error("Error in config file(s): %s", context.appconfigfiles")
```